src/NServiceBus.Core/Routing/FileBasedDynamicRouting/RoutingFileAccess.cs

```diff
@@ -0,0 +1,23 @@
+namespace NServiceBus
+{
+    using System.IO;
+    using System.Threading.Tasks;
+    using System.Xml;
+    using System.Xml.Linq;
+
+    class RoutingFileAccess : IRoutingFileAccess
+    {
+        public Task<XDocument> Load(string path)
+        {
+            XDocument doc;
+            using (var file = new FileStream(path, FileMode.Open, FileAccess.Read, FileShare.ReadWrite))
```

Comment:
For my on education I googled XDocument and async. The only thing I found was this

```
using (XmlReader r = XmlReader.Create(new StringReader(input), new XmlReaderSettings() { Async = true }))
            {
                while (await r.ReadAsync())
                {
                    switch (r.NodeType)
                    {
                        case XmlNodeType.Element:
                            Console.WriteLine(r.LocalName);
                            break;

                        case XmlNodeType.Text:
                            Console.WriteLine(await r.GetValueAsync());
                            break;
                    }
                }
            }
```